### ISSUE TITLE
Unable to install latest version from pip on macOS

### ISSUE THREAD

**Issue by jstncno** [dhermes/bezier#282]
When trying to `pip install bezier==2021.2.12` on macOS Monterey 12.1, I get the following message:

```
$ python3 -m pip install bezier==2021.2.12
Collecting bezier==2021.2.12
  Using cached bezier-2021.2.12.tar.gz (313 kB)
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'python3' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-fmvb02fv/bezier_ae18b1cd5a094f5bb4f810f16d13144f/setup.py'"'"'; __file__='"'"'/private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-fmvb02fv/bezier_ae18b1cd5a094f5bb4f810f16d13144f/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-pip-egg-info-4npnjt5d
       cwd: /private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-fmvb02fv/bezier_ae18b1cd5a094f5bb4f810f16d13144f/
  Complete output (1 lines):
  The BEZIER_INSTALL_PREFIX environment variable must be set.
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/cb/e8/0073ac3f92cbe8ced10e5e5822594a8f63b9a3dc91258c681b7ee96b80b0/bezier-2021.2.12.tar.gz#sha256=d0f752aeb420057ab761f962259c7ca00d925750e324c9bb729f2f10ad39529a (from https://pypi.org/simple/bezier/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement bezier==2021.2.12 (from versions: 0.0.1, 0.1.0, 0.1.1, 0.2.0, 0.2.1, 0.3.0, 0.4.0, 0.5.0, 0.6.0, 0.6.1, 0.6.2, 0.6.3, 0.6.4, 0.7.0, 0.8.0, 0.9.0, 0.10.0, 0.11.0, 2020.1.14, 2020.2.3, 2020.5.19, 2021.2.12)
ERROR: No matching distribution found for bezier==2021.2.12
```

When I don't specify a version, `pip` will try to install previous versions until it finds something it can recognize (in my case its `bezier==2020.1.14`):

```
$ pip install bezier
Collecting bezier
  Using cached bezier-2021.2.12.tar.gz (313 kB)
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'python3.9' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_3ac116677db34f4c81667ca2339f8caf/setup.py'"'"'; __file__='"'"'/private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_3ac116677db34f4c81667ca2339f8caf/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-pip-egg-info-ob3i7c9w
       cwd: /private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_3ac116677db34f4c81667ca2339f8caf/
  Complete output (1 lines):
  The BEZIER_INSTALL_PREFIX environment variable must be set.
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/cb/e8/0073ac3f92cbe8ced10e5e5822594a8f63b9a3dc91258c681b7ee96b80b0/bezier-2021.2.12.tar.gz#sha256=d0f752aeb420057ab761f962259c7ca00d925750e324c9bb729f2f10ad39529a (from https://pypi.org/simple/bezier/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached bezier-2020.5.19.tar.gz (313 kB)
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'python3.9' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_d7df07d25c634be1b5013f7a270a8058/setup.py'"'"'; __file__='"'"'/private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_d7df07d25c634be1b5013f7a270a8058/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-pip-egg-info-cov24fd9
       cwd: /private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_d7df07d25c634be1b5013f7a270a8058/
  Complete output (1 lines):
  The BEZIER_INSTALL_PREFIX environment variable must be set.
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/b9/0c/b1982f93c36fc06850a3880bb693bf24a047946633ca7403664836b9da9e/bezier-2020.5.19.tar.gz#sha256=ff75337e5bb4385376aeaa48fc941bcd61b528df43a1e938bfb07fe7d924b19c (from https://pypi.org/simple/bezier/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached bezier-2020.2.3.tar.gz (310 kB)
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'python3.9' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_31834a8b81f4496dae5f43bb99caaf15/setup.py'"'"'; __file__='"'"'/private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_31834a8b81f4496dae5f43bb99caaf15/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-pip-egg-info-xcd72hnb
       cwd: /private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_31834a8b81f4496dae5f43bb99caaf15/
  Complete output (1 lines):
  The BEZIER_INSTALL_PREFIX environment variable must be set.
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/9a/cb/4859a29397ba6d57ddabc54c03b65e85a86cbf77cb898ad1bb2cfab14c0c/bezier-2020.2.3.tar.gz#sha256=0526631e41ac0416d4dd6be66105670bb1c1a342d4e7e82b3e9009231ac0e868 (from https://pypi.org/simple/bezier/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached bezier-2020.1.14.tar.gz (376 kB)
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'python3.9' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_4eeecc9b190d4cc7b64901cc279f7121/setup.py'"'"'; __file__='"'"'/private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_4eeecc9b190d4cc7b64901cc279f7121/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-pip-egg-info-g7_26w2r
       cwd: /private/var/folders/gn/72hdtdqx06s1vzt_xp8npzw40000gn/T/pip-install-v9vknona/bezier_4eeecc9b190d4cc7b64901cc279f7121/
  Complete output (7 lines):
  Error: NumPy needs to be installed first. It can be installed via:
  
  $ python    -m pip install numpy
  $ python3.8 -m pip install numpy
  $ # OR
  $ conda install numpy
  
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/fa/b9/2b3cca5da2e182d093fbbea8074810da55e9dca2edf96fec94ae6f33a884/bezier-2020.1.14.tar.gz#sha256=74178eea1f6b69d83d02a12311cd8500b5b56ce87fab2a10c2f35d01344102b8 (from https://pypi.org/simple/bezier/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

More info

```
$ python3 -c "import os; print(os.name)"
posix
$ python3 -V
Python 3.9.9
$ python3 -m pip -V                     
pip 21.3.1 from /...
```

**Comment by dhermes** [comment]
Are you using Python 3.10? If yes, see #283.

TL;DR is I haven't cut a `bezier` release since Python 3.10 was released.

**Comment by jstncno** [comment]
Hello, apologies for the delayed response.

I am using Python 3.9.9, but #283 sounds related. Thanks for looking into this!

**Comment by dhermes** [comment]
@jstncno I think the issue here is lack of ARM support in my wheels. Can you confirm if you are running an M1? (I don't have access to one yet, I am still working on sorting that out.)

You can "find" out by running `file /usr/local/bin/python3.9` (or a different path to your binary) and running this in Python

```python
In [1]: import platform

In [2]: platform.machine()
Out[2]: 'x86_64'
```

**Comment by behackl** [comment]
Hey! I have an M1 mac and ran into the same issue over at https://github.com/ManimCommunity/manim/pull/2421 -- is there anything I can do to help out here?

**Comment by jstncno** [comment]
> ```python
> rm.machine()
> Out[2]: 'x86_64'
> ```

```
$ python3
...
>>> import platform
>>> platform.machine()
'arm64'
>>>
```

@dhermes  hope that helps!

**Comment by jstncno** [comment]
> @jstncno I think the issue here is lack of ARM support in my wheels. Can you confirm if you are running an M1? (I don't have access to one yet, I am still working on sorting that out.)
> 
> You can "find" out by running `file /usr/local/bin/python3.9` (or a different path to your binary) and running this in Python
> 
> ```python
> In [1]: import platform
> 
> In [2]: platform.machine()
> Out[2]: 'x86_64'
> ```

**Comment by Darylgolden** [comment]
We (@ManimCommunity) have managed to get an ARM Linux server instance through Oracle. If this fits your needs and you provide a way for us to securely send the SSH details to you, we can let you use it temporarily to create your ARM wheels.

**Comment by dhermes** [comment]
@Darylgolden This issue is about macOS / Darwin ARM support unfortunately. I have friends with an M1 that I can just send instructions to, I am unfortunately very busy these days and haven't had time to do this. I will try tomorrow actually during my work day and see if I can make progress.

**Comment by Darylgolden** [comment]
Hi, any updates on this?

**Comment by dhermes** [comment]
This is resolved! I just released https://pypi.org/project/bezier/2023.7.28/

I'm happy to re-open the discussion if there are further issues, but preemptively closing for now.
